static/aggregate_test.go

```diff
@@ -0,0 +1,150 @@
+	. "github.com/onsi/gomega"
+	"golang.org/x/tools/go/packages"
+)
+
+var _ = Describe("func FromPackages() (aggregate analysis)", func() {
+	When("the application contains a single aggregate handler", func() {
+		It("returns the aggregate handler configuration", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/single-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+			Expect(apps).To(HaveLen(1))
+			Expect(apps[0].Handlers().Aggregates()).To(HaveLen(1))
+
+			a := apps[0].Handlers().Aggregates()[0]
+			Expect(a.Identity()).To(
+				Equal(
+					configkit.Identity{
+						Name: "<aggregate>",
+						Key:  "ef16c9d1-d7b6-4c99-a0e7-a59218e544fc",
+					},
+				),
+			)
+			Expect(a.TypeName()).To(
+				Equal(
+					"github.com/dogmatiq/configkit/static/testdata/aggregates/single-aggregate-app.AggregateHandler",
+				),
+			)
+			Expect(a.HandlerType()).To(Equal(configkit.AggregateHandlerType))
+
+			Expect(a.MessageNames()).To(Equal(
+				configkit.EntityMessageNames{
+					Consumed: message.NameRoles{
+						cfixtures.MessageATypeName: message.CommandRole,
+						cfixtures.MessageBTypeName: message.CommandRole,
+					},
+					Produced: message.NameRoles{
+						cfixtures.MessageCTypeName: message.EventRole,
+						cfixtures.MessageDTypeName: message.EventRole,
+					},
+				},
+			))
+		})
+	})
+
+	When("the application contains multiple aggregate handlers", func() {
+		It("returns all of the aggregate handler configurations", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/multiple-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+			Expect(apps).To(HaveLen(1))
+			Expect(apps[0].Handlers().Aggregates()).To(HaveLen(2))
+
+			var identities []configkit.Identity
+			for _, a := range apps[0].Handlers().Aggregates() {
+				identities = append(identities, a.Identity())
+			}
+
+			Expect(identities).To(
+				ConsistOf(
+					configkit.Identity{
+						Name: "<first-aggregate>",
+						Key:  "e6300d8d-6530-405e-9729-e9ca21df23d3",
+					},
+					configkit.Identity{
+						Name: "<second-aggregate>",
+						Key:  "feeb96d0-c56b-4e58-9cd0-d393683c2ec7",
+					},
+				),
+			)
+		})
+	})
+
+	When("a nil value passed as an aggregate handler", func() {
+		It("does not returns the aggregate handler configuration", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/nil-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+			Expect(apps).To(HaveLen(1))
+			Expect(apps[0].Handlers()).To(Equal(configkit.HandlerSet{}))
+		})
+	})
+
+	When("a nil value passed as a message while configuring an aggregate handler", func() {
```

Comment:
```suggestion
	When("a nil value is passed as a message", func() {
```

Does this test cover both commands and events?

Reply:
Do we need a separate test here for commands vs events?

Reply:
It does cover both commands and events at the moment all in one test, I can split it quite easily though.

Reply:
Nah, all good.. I think this is good to go man!